src/cargo/util/config.rs

```diff
@@ -506,7 +506,11 @@ impl Config {
     }
 
     pub fn network_allowed(&self) -> bool {
-        !self.frozen
+        !self.frozen() && !self.cli_unstable().offline
```

Comment:
I'm a little worried about modifying this function which is already uses in a few other locations. For example I think [this error message](https://github.com/rust-lang/cargo/blob/510cf5eb754ad0fc819796b4dbe9d46cf1db7838/src/cargo/ops/lockfile.rs#L78-L80) may no longer be right (or [this one](https://github.com/rust-lang/cargo/blob/510cf5eb754ad0fc819796b4dbe9d46cf1db7838/src/cargo/ops/registry.rs#L266-L269) or [this one](https://github.com/rust-lang/cargo/blob/510cf5eb754ad0fc819796b4dbe9d46cf1db7838/src/cargo/sources/git/utils.rs#L562-L565)). Perhaps these locations could use a central method to determine why the network isn't allowed?

Reply:
Also I think that these error messages should never happen during offline mode, right? In that offline mode should divert Cargo from executing code paths like that.